src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+export type Color = 'red' | 'green' | 'blue';
+export type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('put correct sides');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('side(s) less than 1');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return +area.toFixed(2);
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
```

Comment:
As with the Triangle class, you can remove the manual property assignment in the constructor since TypeScript will handle it automatically with the 'public' access modifier.